corehq/apps/locations/models.py

```diff
@@ -846,24 +867,30 @@ def set_location_type(self, location_type_name):
             raise LocationType.DoesNotExist(msg)
 
     @classmethod
-    def _migration_get_fields(cls):
+    def get_sync_fields(cls):
         return ["domain", "name", "site_code", "external_id", "metadata",
                 "is_archived", "latitude", "longitude"]
 
-    @classmethod
-    def _migration_get_sql_model_class(cls):
-        return SQLLocation
+    def sync_sql_location(self):
+        if not self._id:
+            self._id = uuid.uuid4().hex
```

Comment:
I guess it doesn't matter too much, but if it was in the save method then you couldn't call sync_sql_location independently, so my approach here was to be explicit about putting it in the context where it was needed. But yeah since we never use sync_sql_location independently I guess it could have also gone in save, thought it wouldn't change the current behavior functionally. I tried to avoid setting it in __init__ because once you have an ._id it kind of implies you have a SQLLocation too so I didn't know if there was any other code that would break if I did that.